packages/backend/src/services/ValidationService/ValidationService.ts

```diff
@@ -0,0 +1,469 @@
+                fieldIds: string[],
+                fieldId: string,
+                error: string,
+            ) => {
+                if (!fieldIds?.includes(fieldId)) {
+                    return [
+                        ...acc,
+                        {
+                            ...commonValidation,
+                            error,
+                        },
+                    ];
+                }
+                return acc;
+            };
+            const dashboardFilterRules = [
+                ...dashboard.filters.dimensions,
+                ...dashboard.filters.metrics,
+            ];
+            const filterErrors = dashboardFilterRules.reduce<
+                CreateValidation[]
+            >(
+                (acc, filter) =>
+                    containsFieldId(
+                        acc,
+                        existingFieldIds,
+                        filter.target.fieldId,
+                        `Filter error: the field '${filter}' no longer exists`,
+                    ),
+                [],
+            );
+
+            const chartTiles = dashboard.tiles.filter(isDashboardChartTileType);
+            const chartErrors = chartTiles.reduce<CreateValidation[]>(
+                (acc, chart) => {
+                    const brokenChart = brokenCharts.find(
+                        (c) => c.chartUuid === chart.properties.savedChartUuid,
+                    );
+                    if (brokenChart !== undefined) {
+                        return [
+                            ...acc,
+                            {
+                                ...commonValidation,
+                                error: `The chart '${brokenChart.name}' is broken on this dashboard.`,
+                            },
+                        ];
+                    }
+                    return acc;
+                },
+                [],
+            );
+            return [...filterErrors, ...chartErrors];
+        });
+
+        return results;
+    }
+
+    async generateValidation(projectUuid: string): Promise<CreateValidation[]> {
+        const explores = await this.projectModel.getExploresFromCache(
+            projectUuid,
+        );
+
+        const existingFields = explores?.reduce<CompiledField[]>(
+            (acc, explore) => {
+                if (!explore.tables) return acc;
+
+                const fields = Object.values(explore.tables).flatMap(
+                    (table) => [
+                        ...Object.values(table.dimensions),
+                        ...Object.values(table.metrics),
+                    ],
+                );
+
+                return [...acc, ...fields];
+            },
+            [],
+        );
+
+        if (!existingFields) {
+            Logger.warn(
+                `No fields found for project validation ${projectUuid}`,
+            );
+            return [];
+        }
+
+        const tableErrors = await ValidationService.validateTables(
```

Comment:
```suggestion
        const tableErrors = await this.validateTables(
```
I think 🤔

Reply:
validateTables is a static method (because it doesn't use any of the class properties) so it can't be called with `this`

Reply:
oops! ignore then